#Stroke Prediction

In [2]:
import pandas as pd
import numpy as np

In [7]:
#Load data
df_str = pd.read_csv('/content/drive/MyDrive/CodingDojo/03-AdvancedML/Week09/Data/healthcare-dataset-stroke-data.csv')

df_str.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

1. Source of data:

  https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

2. Brief description of data:

  According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths.
  This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status

3. What is the target?  

  stroke: 1 if the patient had a stroke or 0 if not

4. What does one row represent? (A person? A business? An event? A product?)

  Each row in the data provides relavant information about the patient.

5. Is this a classification or regression problem?

  Classification

In [10]:
df_str.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


6. How many features does the data have?

  12 columns/features

7. How many rows are in the dataset?

  5110

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Possible imbalance of data and finding the right techqinues/parameters to model an imbalanced dataset correctly.
Deciding the correct standardization, imputation and modeling strategies to use to get the most accurate predictions.

#Adult Income

In [11]:
#Load data
df_inc = pd.read_csv('/content/drive/MyDrive/CodingDojo/03-AdvancedML/Week09/Data/adult.csv')

df_inc.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

1. Source of data:

  https://www.kaggle.com/datasets/wenruliu/adult-income-dataset

2. Brief description of data:

  An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.
  This is a widely cited KNN dataset, useful in predicting income level based on the individual’s personal information.

3. What is the target?  

  Income: <=50K or >50K

4. What does one row represent? (A person? A business? An event? A product?)

  Each row in the data represents a person and their information

5. Is this a classification or regression problem?

  Classification

In [12]:
df_inc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


6. How many features does the data have?

  15 columns/features

7. How many rows are in the dataset?

  48842

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Possible imbalance of data and finding the right techqinues/parameters to model an imbalanced dataset correctly.
Deciding the correct standardization, imputation and modeling strategies to use to get the most accurate predictions.